File: src/main/java/br/anhembi/funmodechild/config/DBUserDetailsService.java
package br.anhembi.funmodechild.config;

import br.anhembi.funmodechild.entity.Customer;
import br.anhembi.funmodechild.repository.CustomerRepository;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;

@Service
public class DBUserDetailsService implements UserDetailsService {

    private final CustomerRepository customerRepository;

    public DBUserDetailsService(CustomerRepository customerRepository) {
        this.customerRepository = customerRepository;
    }

    @Override
    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
        Customer user = customerRepository.findByEmail(email);
        if (user == null) {
            throw new UsernameNotFoundException("No user found with username: " + email);
        }

        return user;
    }
}

File: src/main/kotlin/br/anhembi/funmodechild/config/DBUserDetailsService.kt
package br.anhembi.funmodechild.config

import br.anhembi.funmodechild.repository.CustomerRepository
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.security.core.userdetails.UsernameNotFoundException
import org.springframework.stereotype.Service

@Service
class DBUserDetailsService(private val customerRepository: CustomerRepository) : UserDetailsService {

    @Throws(UsernameNotFoundException::class)
    override fun loadUserByUsername(email: String): UserDetails {
        val user = customerRepository.findByEmail(email)
            ?: throw UsernameNotFoundException("No user found with username: $email")

        return user.toLoggedUser()
    }
}
